user/serializers.py

```diff
@@ -1 +1,29 @@
-# write your code here
+from django.contrib.auth import get_user_model
+from rest_framework import serializers
+
+
+class UserSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = get_user_model()
+        fields = ["id", "username", "email", "password", "is_staff"]
+        read_only_fields = ["id", "is_staff"]
+        extra_kwargs = {
+            "password": {
+                "write_only": True,
+                "min_length": 5,
+            }
+        }
+
+    def create(self, validated_data):
+        return get_user_model().objects.create_user(**validated_data)
+
+    def update(self, instance, validated_data):
+        password = validated_data.pop("password", None)
+        user = super().update(instance, validated_data)
+
+        if password:
+            user.set_password(password)
+
+        user.save()
```

Comment:
it's better to save user in the if-statement

Reply:
I think it's bad idea, because we will not be able to change other user data without changing his password.

Reply:
but according to your code: if there is no password, then the user is updated anyway... It seems not OK

Reply:
`if password: user.set_password`
It's not password check. It's just update... You can write other password, not valid, and user password will be update